perception/autoware_multi_object_tracker/lib/tracker/model/tracker_base.cpp

```diff
@@ -192,6 +240,71 @@ bool Tracker::updateWithoutMeasurement(const rclcpp::Time & timestamp)
   return true;
 }
 
+bool Tracker::createPseudoMeasurement(
+  const types::DynamicObject & meas, types::DynamicObject & pred,
+  const autoware_perception_msgs::msg::Shape & tracker_shape, const bool enlarge_covariance)
+{
+  // Apply linear fall‑off weight on dist square
+  const double dx = meas.pose.position.x - pred.pose.position.x;
+  const double dy = meas.pose.position.y - pred.pose.position.y;
+  const double dist2 = dx * dx + dy * dy;
+  constexpr double d_max_square_inv = 1 / 2.0;  // saturate when distance overs 1.414 m
+  constexpr double min_w = 0.0;
+  const double w_pose = std::clamp(1.0 - dist2 * d_max_square_inv, min_w, 1.0);
+
+  // Blend position
+  pred.pose.position.x = pred.pose.position.x * (1 - w_pose) + meas.pose.position.x * w_pose;
+  pred.pose.position.y = pred.pose.position.y * (1 - w_pose) + meas.pose.position.y * w_pose;
+
+  // Use smoothed shape and its area
+  pred.shape = tracker_shape;
+  pred.area = types::getArea(tracker_shape);
+
+  // Blend orientation
+  if (meas.kinematics.orientation_availability != types::OrientationAvailability::UNAVAILABLE) {
+    double yaw_pred = tf2::getYaw(pred.pose.orientation);
+    double yaw_meas = tf2::getYaw(meas.pose.orientation);
+
+    // Handle SIGN_UNKNOWN: limit yaw difference to [-90°, 90°] to prevent sudden rotations
+    if (meas.kinematics.orientation_availability == types::OrientationAvailability::SIGN_UNKNOWN) {
+      double yaw_diff = yaw_meas - yaw_pred;
+      // Normalize yaw_diff to [-π, π] using fmod
+      yaw_diff = std::fmod(yaw_diff + M_PI, 2 * M_PI) - M_PI;
+      if (yaw_diff > M_PI_2) {
+        yaw_diff -= M_PI;
+      } else if (yaw_diff < -M_PI_2) {
+        yaw_diff += M_PI;
+      }
+      yaw_meas = yaw_pred + yaw_diff;
+    }
+
+    double yaw_fused = yaw_pred * (1 - w_pose) + yaw_meas * w_pose;
```

Comment:
Even for the orientation is available, normalization of yaw difference to [-pi, pi] necessary

```suggestion
    double yaw_diff = yaw_meas - yaw_pred;
    // Normalize yaw_diff to [-π, π] using fmod
    yaw_diff = std::fmod(yaw_diff + M_PI, 2 * M_PI) - M_PI;

    // Handle SIGN_UNKNOWN: limit yaw difference to [-90°, 90°]
    if (meas.kinematics.orientation_availability == types::OrientationAvailability::SIGN_UNKNOWN) {
      if (yaw_diff > M_PI_2) {
        yaw_diff -= M_PI;
      } else if (yaw_diff < -M_PI_2) {
        yaw_diff += M_PI;
      }
    }

    double yaw_fused = yaw_pred + yaw_diff * w_pose;
```